src/main/java/org/simplify4u/plugins/ArtifactResolver.java

```diff
@@ -248,9 +257,50 @@ private Artifact resolveSignature(Artifact artifact, SignatureRequirement requir
                 }
             }
         }
+        if (transitive) {
+            // Transitive dependencies are not compiled/tested, so set requirement to scopes relevant at run-time.
+            // Note: this avoids issues with test-scoped artifacts that are not available to the public, such as
+            //       NullAway's test libraries.
```

Comment:
i think that is special situation only for plugins dependency.
please add - in plugin dependency case